### ISSUE TITLE
Too many OFFSET_OUT_OF_RANGE errors

### ISSUE THREAD

**Issue by paambaati** [tulios/kafkajs#149]
I have a 5-broker Kafka cluster with ~100 topics, and I'm using KafkaJS to subscribe to 2 topics (1 with 400 partitions and 1 with 10), and I'm always stuck at `"The requested offset is not within the range of offsets maintained by the server"` errors.

These errors occur in an endless loop and I'm not able to consume any messages. I've also tried resetting the consumer offsets between runs, and also the `fromBeginning` option.

I know this is very little information, but can you help me understand what circumstances can cause this error? I can provide more info about my setup and code.

**Comment by paambaati** [comment]
I've redacted broker node address and Buffer data, and here's a debug trace —

```
{"level":"DEBUG","timestamp":"2018-10-05T09:30:31.483Z","logger":"kafkajs","message":"[Connection] Connecting","broker":"kafka1.com:9092","clientId":"kafka-reader-local","ssl":false,"sasl":false}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:32.112Z","logger":"kafkajs","message":"[Connection] Request ApiVersions(key: 18, version: 0)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":41}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:32.386Z","logger":"kafkajs","message":"[Connection] Response ApiVersions(key: 18, version: 0)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":268,"data":{"errorCode":0,"apiVersions":[{"apiKey":0,"minVersion":0,"maxVersion":5},{"apiKey":1,"minVersion":0,"maxVersion":7},{"apiKey":2,"minVersion":0,"maxVersion":2},{"apiKey":3,"minVersion":0,"maxVersion":5},{"apiKey":4,"minVersion":0,"maxVersion":1},{"apiKey":5,"minVersion":0,"maxVersion":0},{"apiKey":6,"minVersion":0,"maxVersion":4},{"apiKey":7,"minVersion":0,"maxVersion":1},{"apiKey":8,"minVersion":0,"maxVersion":3},{"apiKey":9,"minVersion":0,"maxVersion":3},{"apiKey":10,"minVersion":0,"maxVersion":1},{"apiKey":11,"minVersion":0,"maxVersion":2},{"apiKey":12,"minVersion":0,"maxVersion":1},{"apiKey":13,"minVersion":0,"maxVersion":1},{"apiKey":14,"minVersion":0,"maxVersion":1},{"apiKey":15,"minVersion":0,"maxVersion":1},{"apiKey":16,"minVersion":0,"maxVersion":1},{"apiKey":17,"minVersion":0,"maxVersion":1},{"apiKey":18,"minVersion":0,"maxVersion":1},{"apiKey":19,"minVersion":0,"maxVersion":2},{"apiKey":20,"minVersion":0,"maxVersion":1},{"apiKey":21,"minVersion":0,"maxVersion":0},{"apiKey":22,"minVersion":0,"maxVersion":0},{"apiKey":23,"minVersion":0,"maxVersion":0},{"apiKey":24,"minVersion":0,"maxVersion":0},{"apiKey":25,"minVersion":0,"maxVersion":0},{"apiKey":26,"minVersion":0,"maxVersion":0},{"apiKey":27,"minVersion":0,"maxVersion":0},{"apiKey":28,"minVersion":0,"maxVersion":0},{"apiKey":29,"minVersion":0,"maxVersion":0},{"apiKey":30,"minVersion":0,"maxVersion":0},{"apiKey":31,"minVersion":0,"maxVersion":0},{"apiKey":32,"minVersion":0,"maxVersion":1},{"apiKey":33,"minVersion":0,"maxVersion":0},{"apiKey":34,"minVersion":0,"maxVersion":0},{"apiKey":35,"minVersion":0,"maxVersion":0},{"apiKey":36,"minVersion":0,"maxVersion":0},{"apiKey":37,"minVersion":0,"maxVersion":0},{"apiKey":38,"minVersion":0,"maxVersion":0},{"apiKey":39,"minVersion":0,"maxVersion":0},{"apiKey":40,"minVersion":0,"maxVersion":0},{"apiKey":41,"minVersion":0,"maxVersion":0},{"apiKey":42,"minVersion":0,"maxVersion":0}]}}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:32.387Z","logger":"kafkajs","message":"[Connection] Connecting","broker":"kafka1.com:9092","clientId":"kafka-reader-local","ssl":false,"sasl":false}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:32.647Z","logger":"kafkajs","message":"[Connection] Request ApiVersions(key: 18, version: 0)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":41}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:32.909Z","logger":"kafkajs","message":"[Connection] Response ApiVersions(key: 18, version: 0)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":268,"data":{"errorCode":0,"apiVersions":[{"apiKey":0,"minVersion":0,"maxVersion":5},{"apiKey":1,"minVersion":0,"maxVersion":7},{"apiKey":2,"minVersion":0,"maxVersion":2},{"apiKey":3,"minVersion":0,"maxVersion":5},{"apiKey":4,"minVersion":0,"maxVersion":1},{"apiKey":5,"minVersion":0,"maxVersion":0},{"apiKey":6,"minVersion":0,"maxVersion":4},{"apiKey":7,"minVersion":0,"maxVersion":1},{"apiKey":8,"minVersion":0,"maxVersion":3},{"apiKey":9,"minVersion":0,"maxVersion":3},{"apiKey":10,"minVersion":0,"maxVersion":1},{"apiKey":11,"minVersion":0,"maxVersion":2},{"apiKey":12,"minVersion":0,"maxVersion":1},{"apiKey":13,"minVersion":0,"maxVersion":1},{"apiKey":14,"minVersion":0,"maxVersion":1},{"apiKey":15,"minVersion":0,"maxVersion":1},{"apiKey":16,"minVersion":0,"maxVersion":1},{"apiKey":17,"minVersion":0,"maxVersion":1},{"apiKey":18,"minVersion":0,"maxVersion":1},{"apiKey":19,"minVersion":0,"maxVersion":2},{"apiKey":20,"minVersion":0,"maxVersion":1},{"apiKey":21,"minVersion":0,"maxVersion":0},{"apiKey":22,"minVersion":0,"maxVersion":0},{"apiKey":23,"minVersion":0,"maxVersion":0},{"apiKey":24,"minVersion":0,"maxVersion":0},{"apiKey":25,"minVersion":0,"maxVersion":0},{"apiKey":26,"minVersion":0,"maxVersion":0},{"apiKey":27,"minVersion":0,"maxVersion":0},{"apiKey":28,"minVersion":0,"maxVersion":0},{"apiKey":29,"minVersion":0,"maxVersion":0},{"apiKey":30,"minVersion":0,"maxVersion":0},{"apiKey":31,"minVersion":0,"maxVersion":0},{"apiKey":32,"minVersion":0,"maxVersion":1},{"apiKey":33,"minVersion":0,"maxVersion":0},{"apiKey":34,"minVersion":0,"maxVersion":0},{"apiKey":35,"minVersion":0,"maxVersion":0},{"apiKey":36,"minVersion":0,"maxVersion":0},{"apiKey":37,"minVersion":0,"maxVersion":0},{"apiKey":38,"minVersion":0,"maxVersion":0},{"apiKey":39,"minVersion":0,"maxVersion":0},{"apiKey":40,"minVersion":0,"maxVersion":0},{"apiKey":41,"minVersion":0,"maxVersion":0},{"apiKey":42,"minVersion":0,"maxVersion":0}]}}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:32.915Z","logger":"kafkajs","message":"[Connection] Request Metadata(key: 3, version: 2)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":1,"size":62}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:33.443Z","logger":"kafkajs","message":"[Connection] Response Metadata(key: 3, version: 2)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":1,"size":13677,"data":{"brokers":[{"nodeId":2,"host":"kafka2.com","port":9092,"rack":null},{"nodeId":5,"host":"kafka5.com","port":9092,"rack":null},{"nodeId":4,"host":"kafka4.com","port":9092,"rack":null},{"nodeId":1,"host":"kafka1.com","port":9092,"rack":null},{"nodeId":3,"host":"kafka3.com","port":9092,"rack":null}],"clusterId":"7QRYYnVdScOZOZEEj9pdpw","controllerId":3,"topicMetadata":[...]}]}}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:33.446Z","logger":"kafkajs","message":"[Connection] Request Metadata(key: 3, version: 2)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":2,"size":82}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:33.731Z","logger":"kafkajs","message":"[Connection] Response Metadata(key: 3, version: 2)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":2,"size":15380,"data":{"brokers":[{"nodeId":2,"host":"kafka2.com","port":9092,"rack":null},{"nodeId":5,"host":"kafka5.com","port":9092,"rack":null},{"nodeId":4,"host":"kafka4.com","port":9092,"rack":null},{"nodeId":1,"host":"kafka1.com","port":9092,"rack":null},{"nodeId":3,"host":"kafka3.com","port":9092,"rack":null}],"clusterId":"7QRYYnVdScOZOZEEj9pdpw","controllerId":3,"topicMetadata":[....]}]}}

{"level":"INFO","timestamp":"2018-10-05T09:30:33.735Z","logger":"kafkajs","message":"[Consumer] Starting","groupId":"kafka-reader-local"}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:33.735Z","logger":"kafkajs","message":"[Connection] Connecting","broker":"kafka3.com:9092","clientId":"kafka-reader-local","ssl":false,"sasl":false}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:33.999Z","logger":"kafkajs","message":"[Connection] Request GroupCoordinator(key: 10, version: 0)","broker":"kafka3.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":70}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:34.261Z","logger":"kafkajs","message":"[Connection] Response GroupCoordinator(key: 10, version: 0)","broker":"kafka3.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":45,"data":{"errorCode":0,"coordinator":{"nodeId":4,"host":"kafka4.com","port":9092}}}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:34.261Z","logger":"kafkajs","message":"[Cluster] Found group coordinator","nodeId":4}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:34.261Z","logger":"kafkajs","message":"[Connection] Connecting","broker":"kafka4.com:9092","clientId":"kafka-reader-local","ssl":false,"sasl":false}

{"level":"DEBUG","timestamp":"2018-10-05T09:30:34.791Z","logger":"kafkajs","message":"[Connection] Request JoinGroup(key: 11, version: 0)","broker":"kafka4.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":161}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:32.754Z","logger":"kafkajs","message":"[Connection] Response JoinGroup(key: 11, version: 0)","broker":"kafka4.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":283,"data":{"errorCode":0,"generationId":35,"groupProtocol":"RoundRobinAssigner","leaderId":"kafka-reader-local-d994b0c2-c6f9-4d63-800c-b3e285b710ab","memberId":"kafka-reader-local-d994b0c2-c6f9-4d63-800c-b3e285b710ab","members":[{"memberId":"kafka-reader-local-d994b0c2-c6f9-4d63-800c-b3e285b710ab","memberMetadata":{"type":"Buffer","data":[.....]}}]}}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:32.755Z","logger":"kafkajs","message":"[ConsumerGroup] Chosen as group leader","groupId":"kafka-reader-local","generationId":35,"memberId":"kafka-reader-local-d994b0c2-c6f9-4d63-800c-b3e285b710ab","topics":["topic1","topic2"]}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:32.758Z","logger":"kafkajs","message":"[ConsumerGroup] Group assignment","groupId":"kafka-reader-local","generationId":35,"topics":["topic1","topic2"],"groupProtocol":"RoundRobinAssigner","assignment":[{"memberId":"kafka-reader-local-d994b0c2-c6f9-4d63-800c-b3e285b710ab","memberAssignment":{"type":"Buffer","data":[....]}}]}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:32.785Z","logger":"kafkajs","message":"[Connection] Request SyncGroup(key: 14, version: 0)","broker":"kafka4.com:9092","clientId":"kafka-reader-local","correlationId":1,"size":2069}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.062Z","logger":"kafkajs","message":"[Connection] Response SyncGroup(key: 14, version: 0)","broker":"kafka4.com:9092","clientId":"kafka-reader-local","correlationId":1,"size":1865,"data":{"errorCode":0,"memberAssignment":{"type":"Buffer","data":[.....]}}}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.063Z","logger":"kafkajs","message":"[ConsumerGroup] Received assignment","groupId":"kafka-reader-local","generationId":35,"memberId":"kafka-reader-local-d994b0c2-c6f9-4d63-800c-b3e285b710ab","memberAssignment":{"topic1":[......]}}

{"level":"INFO","timestamp":"2018-10-05T09:32:33.063Z","logger":"kafkajs","message":"[Runner] Consumer has joined the group","groupId":"kafka-reader-local","memberId":"kafka-reader-local-d994b0c2-c6f9-4d63-800c-b3e285b710ab","leaderId":"kafka-reader-local-d994b0c2-c6f9-4d63-800c-b3e285b710ab","isLeader":true,"duration":119328}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.073Z","logger":"kafkajs","message":"[Connection] Request OffsetFetch(key: 9, version: 2)","broker":"kafka4.com:9092","clientId":"kafka-reader-local","correlationId":2,"size":1919}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.343Z","logger":"kafkajs","message":"[Connection] Response OffsetFetch(key: 9, version: 2)","broker":"kafka4.com:9092","clientId":"kafka-reader-local","correlationId":2,"size":7255,"data":{.....}}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.351Z","logger":"kafkajs","message":"[Connection] Connecting","broker":"kafka2.com:9092","clientId":"kafka-reader-local","ssl":false,"sasl":false}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.351Z","logger":"kafkajs","message":"[Connection] Connecting","broker":"kafka5.com:9092","clientId":"kafka-reader-local","ssl":false,"sasl":false}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.359Z","logger":"kafkajs","message":"[Connection] Request ListOffsets(key: 2, version: 0)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":3,"size":86}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.359Z","logger":"kafkajs","message":"[Connection] Request ListOffsets(key: 2, version: 0)","broker":"kafka3.com:9092","clientId":"kafka-reader-local","correlationId":1,"size":118}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.359Z","logger":"kafkajs","message":"[Connection] Request ListOffsets(key: 2, version: 0)","broker":"kafka4.com:9092","clientId":"kafka-reader-local","correlationId":3,"size":102}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.618Z","logger":"kafkajs","message":"[Connection] Request ListOffsets(key: 2, version: 0)","broker":"kafka5.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":86}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.618Z","logger":"kafkajs","message":"[Connection] Response ListOffsets(key: 2, version: 0)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":3,"size":47,"data":{"responses":[{"topic":"topic1","partitions":[{"partition":110,"errorCode":0,"offsets":["4257512"]}]}]}}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.619Z","logger":"kafkajs","message":"[Connection] Response ListOffsets(key: 2, version: 0)","broker":"kafka3.com:9092","clientId":"kafka-reader-local","correlationId":1,"size":83,"data":{"responses":[{"topic":"topic1","partitions":[{"partition":102,"errorCode":0,"offsets":["3978846"]},{"partition":92,"errorCode":0,"offsets":["4129268"]},{"partition":77,"errorCode":0,"offsets":["4269084"]}]}]}}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.619Z","logger":"kafkajs","message":"[Connection] Response ListOffsets(key: 2, version: 0)","broker":"kafka4.com:9092","clientId":"kafka-reader-local","correlationId":3,"size":65,"data":{"responses":[{"topic":"topic1","partitions":[{"partition":128,"errorCode":0,"offsets":["4016637"]},{"partition":158,"errorCode":0,"offsets":["4366767"]}]}]}}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.878Z","logger":"kafkajs","message":"[Connection] Response ListOffsets(key: 2, version: 0)","broker":"kafka5.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":47,"data":{"responses":[{"topic":"topic1","partitions":[{"partition":29,"errorCode":0,"offsets":["4285790"]}]}]}}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:33.883Z","logger":"kafkajs","message":"[Connection] Request ListOffsets(key: 2, version: 0)","broker":"kafka2.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":118}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:34.144Z","logger":"kafkajs","message":"[Connection] Response ListOffsets(key: 2, version: 0)","broker":"kafka2.com:9092","clientId":"kafka-reader-local","correlationId":0,"size":83,"data":{"responses":[{"topic":"topic1","partitions":[{"partition":226,"errorCode":0,"offsets":["4272587"]},{"partition":86,"errorCode":0,"offsets":["3881085"]},{"partition":346,"errorCode":0,"offsets":["3913838"]}]}]}}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:34.152Z","logger":"kafkajs","message":"[ConsumerGroup] Fetching from 2 out of 2 topics","topics":["topic1","topic2"],"activeTopics":["topic1","topic2"],"pausedTopics":[]}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:34.324Z","logger":"kafkajs","message":"[Connection] Request Fetch(key: 1, version: 3)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","correlationId":4,"size":1546}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:34.324Z","logger":"kafkajs","message":"[Connection] Request Fetch(key: 1, version: 3)","broker":"kafka2.com:9092","clientId":"kafka-reader-local","correlationId":1,"size":1546}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:34.324Z","logger":"kafkajs","message":"[Connection] Request Fetch(key: 1, version: 3)","broker":"kafka3.com:9092","clientId":"kafka-reader-local","correlationId":2,"size":1546}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:34.324Z","logger":"kafkajs","message":"[Connection] Request Fetch(key: 1, version: 3)","broker":"kafka4.com:9092","clientId":"kafka-reader-local","correlationId":4,"size":1546}

{"level":"DEBUG","timestamp":"2018-10-05T09:32:34.325Z","logger":"kafkajs","message":"[Connection] Request Fetch(key: 1, version: 3)","broker":"kafka5.com:9092","clientId":"kafka-reader-local","correlationId":1,"size":1546}

{"level":"ERROR","timestamp":"2018-10-05T09:33:56.435Z","logger":"kafkajs","message":"[Connection] Response Fetch(key: 1, version: 3)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","error":"The requested offset is not within the range of offsets maintained by the server","correlationId":4,"size":20349652}

{"level":"DEBUG","timestamp":"2018-10-05T09:33:56.435Z","logger":"kafkajs","message":"[Connection] Response Fetch(key: 1, version: 3)","broker":"kafka1.com:9092","clientId":"kafka-reader-local","error":"The requested offset is not within the range of offsets maintained by the server","correlationId":4,"payload":{"type":"Buffer","data":

{"level":"ERROR","timestamp":"2018-10-05T09:54:25.481Z","logger":"kafkajs","message":"[ConsumerGroup] Offset out of range, resetting to default offset","topic":"topic1","partition":355,"groupId":"kafka-reader-local","memberId":"kafka-reader-local-d994b0c2-c6f9-4d63-800c-b3e285b710ab"}
```

It is also available as a gist — https://gist.github.com/paambaati/7ee798fc2af5c8bc397299bdb07b8b33#file-kafkajs-log

**Comment by tulios** [comment]
Sure, let's debug this.
This error can happen in two situations:
1) It's the first time the consumer group is fetching the topic, so it has to apply the default offset (`fromBeginning`)
2) The consumer group hasn't been active for a while and the offsets "expired", which forces the consumer group to apply the default offset logic again

Do you know if the `groupId` is unique? no other consumer is active with the same id
Do you know if this is a compacted topic?
Have you tried subscribing to only one topic? Same issue?

We have been using KafkaJS with topics around 50 partitions, it's the first time I see 400 partition topics, but it should work just fine. I've set up a local environment with 5 nodes, and 1 topic with 400 partitions and I can consume just fine. I think this is a compacted topic, can you confirm?

**Comment by paambaati** [comment]
> Do you know if the groupId is unique? no other consumer is active with the same id

It is, yes.

> Do you know if this is a compacted topic?

It is not. The topic's default cleanup policy is `delete`.

> Have you tried subscribing to only one topic? Same issue?

I did, yes. I still see the same issue.

@tulios Also, a quick question about this —

>  apply the default offset 

If the topic is pretty huge in size (say ~4 GB with 1.6B summed offsets), is that a possible bottleneck? What exactly happens when applying the default offset? I also tried the same thing with `fromBeginning: false`, but I still see it happening. Also, FWIW, I'm running Kafka 1.1.1.

**Comment by paambaati** [comment]
@tulios `Response OffsetFetch` —


```
{"level":"DEBUG","timestamp":"2018-10-08T07:14:02.330Z","logger":"kafkajs","message":"[Connection] Response OffsetFetch(key: 9, version: 2)","broker":"kafka4.com:9092","clientId":"kafka-reader-local","correlationId":2,"size":6431,"data":{"responses":[{"topic":"topic1","partitions":[{"partition":0,"offset":"4143806","metadata":"","errorCode":0},{"partition":1,"offset":"4543306","metadata":"","errorCode":0},{"partition":2,"offset":"4239347","metadata":"","errorCode":0},{"partition":3,"offset":"4239104","metadata":"","errorCode":0},{"partition":4,"offset":"4094933","metadata":"","errorCode":0},{"partition":5,"offset":"4241632","metadata":"","errorCode":0},{"partition":6,"offset":"3966991","metadata":"","errorCode":0},{"partition":7,"offset":"4169178","metadata":"","errorCode":0},{"partition":8,"offset":"3905377","metadata":"","errorCode":0},{"partition":9,"offset":"4296918","metadata":"","errorCode":0},{"partition":10,"offset":"4065167","metadata":"","errorCode":0},{"partition":11,"offset":"4043429","metadata":"","errorCode":0},{"partition":12,"offset":"4326236","metadata":"","errorCode":0},{"partition":13,"offset":"4176365","metadata":"","errorCode":0},{"partition":14,"offset":"4454786","metadata":"","errorCode":0},{"partition":15,"offset":"4378623","metadata":"","errorCode":0},{"partition":16,"offset":"4195897","metadata":"","errorCode":0},{"partition":17,"offset":"4321289","metadata":"","errorCode":0},{"partition":18,"offset":"5046350","metadata":"","errorCode":0},{"partition":19,"offset":"4056064","metadata":"","errorCode":0},{"partition":20,"offset":"4277506","metadata":"","errorCode":0},{"partition":21,"offset":"4108382","metadata":"","errorCode":0},{"partition":22,"offset":"4070376","metadata":"","errorCode":0},{"partition":23,"offset":"4135697","metadata":"","errorCode":0},{"partition":24,"offset":"4246050","metadata":"","errorCode":0},{"partition":25,"offset":"4014287","metadata":"","errorCode":0},{"partition":26,"offset":"4017851","metadata":"","errorCode":0},{"partition":27,"offset":"4214520","metadata":"","errorCode":0},{"partition":28,"offset":"4264264","metadata":"","errorCode":0},{"partition":29,"offset":"4398252","metadata":"","errorCode":0},{"partition":30,"offset":"4367232","metadata":"","errorCode":0},{"partition":31,"offset":"4311537","metadata":"","errorCode":0},{"partition":32,"offset":"4158827","metadata":"","errorCode":0},{"partition":33,"offset":"4062001","metadata":"","errorCode":0},{"partition":34,"offset":"4221490","metadata":"","errorCode":0},{"partition":35,"offset":"4194665","metadata":"","errorCode":0},{"partition":36,"offset":"4218629","metadata":"","errorCode":0},{"partition":37,"offset":"3959108","metadata":"","errorCode":0},{"partition":38,"offset":"4229934","metadata":"","errorCode":0},{"partition":39,"offset":"4182472","metadata":"","errorCode":0},{"partition":40,"offset":"3939164","metadata":"","errorCode":0},{"partition":41,"offset":"4005374","metadata":"","errorCode":0},{"partition":42,"offset":"4035521","metadata":"","errorCode":0},{"partition":43,"offset":"4031915","metadata":"","errorCode":0},{"partition":44,"offset":"4126694","metadata":"","errorCode":0},{"partition":45,"offset":"4686288","metadata":"","errorCode":0},{"partition":46,"offset":"4246617","metadata":"","errorCode":0},{"partition":47,"offset":"4531681","metadata":"","errorCode":0},{"partition":48,"offset":"4353415","metadata":"","errorCode":0},{"partition":49,"offset":"4696009","metadata":"","errorCode":0},{"partition":50,"offset":"5139969","metadata":"","errorCode":0},{"partition":51,"offset":"4330594","metadata":"","errorCode":0},{"partition":52,"offset":"4292130","metadata":"","errorCode":0},{"partition":53,"offset":"4312174","metadata":"","errorCode":0},{"partition":54,"offset":"4358696","metadata":"","errorCode":0},{"partition":55,"offset":"4286961","metadata":"","errorCode":0},{"partition":56,"offset":"4395033","metadata":"","errorCode":0},{"partition":57,"offset":"4386532","metadata":"","errorCode":0},{"partition":58,"offset":"4363004","metadata":"","errorCode":0},{"partition":59,"offset":"4384583","metadata":"","errorCode":0},{"partition":60,"offset":"4165821","metadata":"","errorCode":0},{"partition":61,"offset":"4411646","metadata":"","errorCode":0},{"partition":62,"offset":"4571086","metadata":"","errorCode":0},{"partition":63,"offset":"4359393","metadata":"","errorCode":0},{"partition":64,"offset":"4468544","metadata":"","errorCode":0},{"partition":65,"offset":"4315484","metadata":"","errorCode":0},{"partition":66,"offset":"4380958","metadata":"","errorCode":0},{"partition":67,"offset":"4298407","metadata":"","errorCode":0},{"partition":68,"offset":"4176735","metadata":"","errorCode":0},{"partition":69,"offset":"4353334","metadata":"","errorCode":0},{"partition":70,"offset":"4232500","metadata":"","errorCode":0},{"partition":71,"offset":"4230631","metadata":"","errorCode":0},{"partition":72,"offset":"4243995","metadata":"","errorCode":0},{"partition":73,"offset":"4261817","metadata":"","errorCode":0},{"partition":74,"offset":"4461557","metadata":"","errorCode":0},{"partition":75,"offset":"4556539","metadata":"","errorCode":0},{"partition":76,"offset":"4363875","metadata":"","errorCode":0},{"partition":77,"offset":"4379406","metadata":"","errorCode":0},{"partition":78,"offset":"4448660","metadata":"","errorCode":0},{"partition":79,"offset":"4882099","metadata":"","errorCode":0},{"partition":80,"offset":"4132075","metadata":"","errorCode":0},{"partition":81,"offset":"4218184","metadata":"","errorCode":0},{"partition":82,"offset":"4444150","metadata":"","errorCode":0},{"partition":83,"offset":"4078831","metadata":"","errorCode":0},{"partition":84,"offset":"4178886","metadata":"","errorCode":0},{"partition":85,"offset":"4186120","metadata":"","errorCode":0},{"partition":86,"offset":"3985108","metadata":"","errorCode":0},{"partition":87,"offset":"4159917","metadata":"","errorCode":0},{"partition":88,"offset":"3918527","metadata":"","errorCode":0},{"partition":89,"offset":"4182537","metadata":"","errorCode":0},{"partition":90,"offset":"4215847","metadata":"","errorCode":0},{"partition":91,"offset":"3948145","metadata":"","errorCode":0},{"partition":92,"offset":"4259462","metadata":"","errorCode":0},{"partition":93,"offset":"4679522","metadata":"","errorCode":0},{"partition":94,"offset":"4257683","metadata":"","errorCode":0},{"partition":95,"offset":"4196914","metadata":"","errorCode":0},{"partition":96,"offset":"4318679","metadata":"","errorCode":0},{"partition":97,"offset":"4118215","metadata":"","errorCode":0},{"partition":98,"offset":"4505640","metadata":"","errorCode":0},{"partition":99,"offset":"4196707","metadata":"","errorCode":0},{"partition":100,"offset":"4305901","metadata":"","errorCode":0},{"partition":101,"offset":"4260613","metadata":"","errorCode":0},{"partition":102,"offset":"4076424","metadata":"","errorCode":0},{"partition":103,"offset":"4758172","metadata":"","errorCode":0},{"partition":104,"offset":"4304107","metadata":"","errorCode":0},{"partition":105,"offset":"4098768","metadata":"","errorCode":0},{"partition":106,"offset":"4278445","metadata":"","errorCode":0},{"partition":107,"offset":"4098085","metadata":"","errorCode":0},{"partition":108,"offset":"4100691","metadata":"","errorCode":0},{"partition":109,"offset":"4153436","metadata":"","errorCode":0},{"partition":110,"offset":"4358286","metadata":"","errorCode":0},{"partition":111,"offset":"4248626","metadata":"","errorCode":0},{"partition":112,"offset":"4195626","metadata":"","errorCode":0},{"partition":113,"offset":"4063304","metadata":"","errorCode":0},{"partition":114,"offset":"4288514","metadata":"","errorCode":0},{"partition":115,"offset":"4178976","metadata":"","errorCode":0},{"partition":116,"offset":"4099960","metadata":"","errorCode":0},{"partition":117,"offset":"3971248","metadata":"","errorCode":0},{"partition":118,"offset":"4206855","metadata":"","errorCode":0},{"partition":119,"offset":"4170588","metadata":"","errorCode":0},{"partition":120,"offset":"4154023","metadata":"","errorCode":0},{"partition":121,"offset":"4175692","metadata":"","errorCode":0},{"partition":122,"offset":"4075610","metadata":"","errorCode":0},{"partition":123,"offset":"4086194","metadata":"","errorCode":0},{"partition":124,"offset":"4254846","metadata":"","errorCode":0},{"partition":125,"offset":"4102456","metadata":"","errorCode":0},{"partition":126,"offset":"4433427","metadata":"","errorCode":0},{"partition":127,"offset":"4219314","metadata":"","errorCode":0},{"partition":128,"offset":"4133069","metadata":"","errorCode":0},{"partition":129,"offset":"4904696","metadata":"","errorCode":0},{"partition":130,"offset":"4424735","metadata":"","errorCode":0},{"partition":131,"offset":"4205430","metadata":"","errorCode":0},{"partition":132,"offset":"4332141","metadata":"","errorCode":0},{"partition":133,"offset":"4577976","metadata":"","errorCode":0},{"partition":134,"offset":"4589327","metadata":"","errorCode":0},{"partition":135,"offset":"4456593","metadata":"","errorCode":0},{"partition":136,"offset":"4315571","metadata":"","errorCode":0},{"partition":137,"offset":"4238763","metadata":"","errorCode":0},{"partition":138,"offset":"4234114","metadata":"","errorCode":0},{"partition":139,"offset":"4463620","metadata":"","errorCode":0},{"partition":140,"offset":"4322010","metadata":"","errorCode":0},{"partition":141,"offset":"4204710","metadata":"","errorCode":0},{"partition":142,"offset":"4408544","metadata":"","errorCode":0},{"partition":143,"offset":"4270629","metadata":"","errorCode":0},{"partition":144,"offset":"4302026","metadata":"","errorCode":0},{"partition":145,"offset":"4602491","metadata":"","errorCode":0},{"partition":146,"offset":"4483332","metadata":"","errorCode":0},{"partition":147,"offset":"4447752","metadata":"","errorCode":0},{"partition":148,"offset":"4277824","metadata":"","errorCode":0},{"partition":149,"offset":"4328324","metadata":"","errorCode":0},{"partition":150,"offset":"4256433","metadata":"","errorCode":0},{"partition":151,"offset":"4365755","metadata":"","errorCode":0},{"partition":152,"offset":"4207938","metadata":"","errorCode":0},{"partition":153,"offset":"4419041","metadata":"","errorCode":0},{"partition":154,"offset":"4286773","metadata":"","errorCode":0},{"partition":155,"offset":"4345355","metadata":"","errorCode":0},{"partition":156,"offset":"4315367","metadata":"","errorCode":0},{"partition":157,"offset":"4357796","metadata":"","errorCode":0},{"partition":158,"offset":"4463591","metadata":"","errorCode":0},{"partition":159,"offset":"4579069","metadata":"","errorCode":0},{"partition":160,"offset":"4190007","metadata":"","errorCode":0},{"partition":161,"offset":"4433907","metadata":"","errorCode":0},{"partition":162,"offset":"4268577","metadata":"","errorCode":0},{"partition":163,"offset":"4250718","metadata":"","errorCode":0},{"partition":164,"offset":"4782866","metadata":"","errorCode":0},{"partition":165,"offset":"4053916","metadata":"","errorCode":0},{"partition":166,"offset":"4064074","metadata":"","errorCode":0},{"partition":167,"offset":"4030281","metadata":"","errorCode":0},{"partition":168,"offset":"4119845","metadata":"","errorCode":0},{"partition":169,"offset":"4104712","metadata":"","errorCode":0},{"partition":170,"offset":"4630939","metadata":"","errorCode":0},{"partition":171,"offset":"3939364","metadata":"","errorCode":0},{"partition":172,"offset":"4118605","metadata":"","errorCode":0},{"partition":173,"offset":"4319904","metadata":"","errorCode":0},{"partition":174,"offset":"4218449","metadata":"","errorCode":0},{"partition":175,"offset":"4081479","metadata":"","errorCode":0},{"partition":176,"offset":"4461242","metadata":"","errorCode":0},{"partition":177,"offset":"4026870","metadata":"","errorCode":0},{"partition":178,"offset":"4327250","metadata":"","errorCode":0},{"partition":179,"offset":"4196801","metadata":"","errorCode":0},{"partition":180,"offset":"4061231","metadata":"","errorCode":0},{"partition":181,"offset":"4214169","metadata":"","errorCode":0},{"partition":182,"offset":"4045195","metadata":"","errorCode":0},{"partition":183,"offset":"4294172","metadata":"","errorCode":0},{"partition":184,"offset":"5021050","metadata":"","errorCode":0},{"partition":185,"offset":"4067222","metadata":"","errorCode":0},{"partition":186,"offset":"4028025","metadata":"","errorCode":0},{"partition":187,"offset":"4296363","metadata":"","errorCode":0},{"partition":188,"offset":"4062582","metadata":"","errorCode":0},{"partition":189,"offset":"4006191","metadata":"","errorCode":0},{"partition":190,"offset":"4357589","metadata":"","errorCode":0},{"partition":191,"offset":"4018866","metadata":"","errorCode":0},{"partition":192,"offset":"4648355","metadata":"","errorCode":0},{"partition":193,"offset":"4020316","metadata":"","errorCode":0},{"partition":194,"offset":"4431831","metadata":"","errorCode":0},{"partition":195,"offset":"4193969","metadata":"","errorCode":0},{"partition":196,"offset":"4144379","metadata":"","errorCode":0},{"partition":197,"offset":"4170125","metadata":"","errorCode":0},{"partition":198,"offset":"4013964","metadata":"","errorCode":0},{"partition":199,"offset":"4266347","metadata":"","errorCode":0},{"partition":200,"offset":"4423347","metadata":"","errorCode":0},{"partition":201,"offset":"4011060","metadata":"","errorCode":0},{"partition":202,"offset":"4017696","metadata":"","errorCode":0},{"partition":203,"offset":"3994815","metadata":"","errorCode":0},{"partition":204,"offset":"4284836","metadata":"","errorCode":0},{"partition":205,"offset":"4219642","metadata":"","errorCode":0},{"partition":206,"offset":"4157424","metadata":"","errorCode":0},{"partition":207,"offset":"4400954","metadata":"","errorCode":0},{"partition":208,"offset":"4110524","metadata":"","errorCode":0},{"partition":209,"offset":"4619305","metadata":"","errorCode":0},{"partition":210,"offset":"4451566","metadata":"","errorCode":0},{"partition":211,"offset":"4403045","metadata":"","errorCode":0},{"partition":212,"offset":"4297510","metadata":"","errorCode":0},{"partition":213,"offset":"4543194","metadata":"","errorCode":0},{"partition":214,"offset":"4264254","metadata":"","errorCode":0},{"partition":215,"offset":"4271298","metadata":"","errorCode":0},{"partition":216,"offset":"4212428","metadata":"","errorCode":0},{"partition":217,"offset":"4372271","metadata":"","errorCode":0},{"partition":218,"offset":"4170314","metadata":"","errorCode":0},{"partition":219,"offset":"4380736","metadata":"","errorCode":0},{"partition":220,"offset":"4167339","metadata":"","errorCode":0},{"partition":221,"offset":"4193126","metadata":"","errorCode":0},{"partition":222,"offset":"4297119","metadata":"","errorCode":0},{"partition":223,"offset":"4307652","metadata":"","errorCode":0},{"partition":224,"offset":"4217688","metadata":"","errorCode":0},{"partition":225,"offset":"4356789","metadata":"","errorCode":0},{"partition":226,"offset":"4373229","metadata":"","errorCode":0},{"partition":227,"offset":"4509992","metadata":"","errorCode":0},{"partition":228,"offset":"4743873","metadata":"","errorCode":0},{"partition":229,"offset":"4853283","metadata":"","errorCode":0},{"partition":230,"offset":"4317316","metadata":"","errorCode":0},{"partition":231,"offset":"4226691","metadata":"","errorCode":0},{"partition":232,"offset":"4508335","metadata":"","errorCode":0},{"partition":233,"offset":"4248126","metadata":"","errorCode":0},{"partition":234,"offset":"4289605","metadata":"","errorCode":0},{"partition":235,"offset":"4329781","metadata":"","errorCode":0},{"partition":236,"offset":"4584730","metadata":"","errorCode":0},{"partition":237,"offset":"4129646","metadata":"","errorCode":0},{"partition":238,"offset":"4630292","metadata":"","errorCode":0},{"partition":239,"offset":"4389623","metadata":"","errorCode":0},{"partition":240,"offset":"4071017","metadata":"","errorCode":0},{"partition":241,"offset":"4353675","metadata":"","errorCode":0},{"partition":242,"offset":"4330661","metadata":"","errorCode":0},{"partition":243,"offset":"4244655","metadata":"","errorCode":0},{"partition":244,"offset":"4272362","metadata":"","errorCode":0},{"partition":245,"offset":"4040136","metadata":"","errorCode":0},{"partition":246,"offset":"4002945","metadata":"","errorCode":0},{"partition":247,"offset":"3995715","metadata":"","errorCode":0},{"partition":248,"offset":"3887859","metadata":"","errorCode":0},{"partition":249,"offset":"4188879","metadata":"","errorCode":0},{"partition":250,"offset":"4301774","metadata":"","errorCode":0},{"partition":251,"offset":"3929527","metadata":"","errorCode":0},{"partition":252,"offset":"4112764","metadata":"","errorCode":0},{"partition":253,"offset":"4168407","metadata":"","errorCode":0},{"partition":254,"offset":"4192349","metadata":"","errorCode":0},{"partition":255,"offset":"4661344","metadata":"","errorCode":0},{"partition":256,"offset":"4309525","metadata":"","errorCode":0},{"partition":257,"offset":"4048270","metadata":"","errorCode":0},{"partition":258,"offset":"4289149","metadata":"","errorCode":0},{"partition":259,"offset":"4148801","metadata":"","errorCode":0},{"partition":260,"offset":"4636987","metadata":"","errorCode":0},{"partition":261,"offset":"3965969","metadata":"","errorCode":0},{"partition":262,"offset":"4007979","metadata":"","errorCode":0},{"partition":263,"offset":"4087887","metadata":"","errorCode":0},{"partition":264,"offset":"4430699","metadata":"","errorCode":0},{"partition":265,"offset":"4048145","metadata":"","errorCode":0},{"partition":266,"offset":"3952356","metadata":"","errorCode":0},{"partition":267,"offset":"4262144","metadata":"","errorCode":0},{"partition":268,"offset":"4282185","metadata":"","errorCode":0},{"partition":269,"offset":"3960399","metadata":"","errorCode":0},{"partition":270,"offset":"4377788","metadata":"","errorCode":0},{"partition":271,"offset":"4081678","metadata":"","errorCode":0},{"partition":272,"offset":"4340216","metadata":"","errorCode":0},{"partition":273,"offset":"4061145","metadata":"","errorCode":0},{"partition":274,"offset":"4366566","metadata":"","errorCode":0},{"partition":275,"offset":"4233358","metadata":"","errorCode":0},{"partition":276,"offset":"4350881","metadata":"","errorCode":0},{"partition":277,"offset":"3910214","metadata":"","errorCode":0},{"partition":278,"offset":"4240990","metadata":"","errorCode":0},{"partition":279,"offset":"4359340","metadata":"","errorCode":0},{"partition":280,"offset":"4042984","metadata":"","errorCode":0},{"partition":281,"offset":"4027849","metadata":"","errorCode":0},{"partition":282,"offset":"4239078","metadata":"","errorCode":0},{"partition":283,"offset":"4033785","metadata":"","errorCode":0},{"partition":284,"offset":"4121837","metadata":"","errorCode":0},{"partition":285,"offset":"4207110","metadata":"","errorCode":0},{"partition":286,"offset":"4249362","metadata":"","errorCode":0},{"partition":287,"offset":"4302953","metadata":"","errorCode":0},{"partition":288,"offset":"4295662","metadata":"","errorCode":0},{"partition":289,"offset":"4673452","metadata":"","errorCode":0},{"partition":290,"offset":"4643085","metadata":"","errorCode":0},{"partition":291,"offset":"4287895","metadata":"","errorCode":0},{"partition":292,"offset":"4247709","metadata":"","errorCode":0},{"partition":293,"offset":"4269601","metadata":"","errorCode":0},{"partition":294,"offset":"4536538","metadata":"","errorCode":0},{"partition":295,"offset":"4295158","metadata":"","errorCode":0},{"partition":296,"offset":"4175888","metadata":"","errorCode":0},{"partition":297,"offset":"4163881","metadata":"","errorCode":0},{"partition":298,"offset":"4209662","metadata":"","errorCode":0},{"partition":299,"offset":"4535307","metadata":"","errorCode":0},{"partition":300,"offset":"4353436","metadata":"","errorCode":0},{"partition":301,"offset":"4422951","metadata":"","errorCode":0},{"partition":302,"offset":"4628994","metadata":"","errorCode":0},{"partition":303,"offset":"4941813","metadata":"","errorCode":0},{"partition":304,"offset":"4209047","metadata":"","errorCode":0},{"partition":305,"offset":"4288423","metadata":"","errorCode":0},{"partition":306,"offset":"4357978","metadata":"","errorCode":0},{"partition":307,"offset":"4268242","metadata":"","errorCode":0},{"partition":308,"offset":"4103082","metadata":"","errorCode":0},{"partition":309,"offset":"4350084","metadata":"","errorCode":0},{"partition":310,"offset":"4171759","metadata":"","errorCode":0},{"partition":311,"offset":"4420239","metadata":"","errorCode":0},{"partition":312,"offset":"4318585","metadata":"","errorCode":0},{"partition":313,"offset":"4924603","metadata":"","errorCode":0},{"partition":314,"offset":"4610633","metadata":"","errorCode":0},{"partition":315,"offset":"4650104","metadata":"","errorCode":0},{"partition":316,"offset":"4397625","metadata":"","errorCode":0},{"partition":317,"offset":"4309193","metadata":"","errorCode":0},{"partition":318,"offset":"4405438","metadata":"","errorCode":0},{"partition":319,"offset":"4622441","metadata":"","errorCode":0},{"partition":320,"offset":"4076969","metadata":"","errorCode":0},{"partition":321,"offset":"4185678","metadata":"","errorCode":0},{"partition":322,"offset":"4323609","metadata":"","errorCode":0},{"partition":323,"offset":"4045732","metadata":"","errorCode":0},{"partition":324,"offset":"4075433","metadata":"","errorCode":0},{"partition":325,"offset":"4040001","metadata":"","errorCode":0},{"partition":326,"offset":"4512583","metadata":"","errorCode":0},{"partition":327,"offset":"3980622","metadata":"","errorCode":0},{"partition":328,"offset":"4059095","metadata":"","errorCode":0},{"partition":329,"offset":"4569165","metadata":"","errorCode":0},{"partition":330,"offset":"4216439","metadata":"","errorCode":0},{"partition":331,"offset":"4014687","metadata":"","errorCode":0},{"partition":332,"offset":"4116875","metadata":"","errorCode":0},{"partition":333,"offset":"4123118","metadata":"","errorCode":0},{"partition":334,"offset":"4192440","metadata":"","errorCode":0},{"partition":335,"offset":"4089810","metadata":"","errorCode":0},{"partition":336,"offset":"4145773","metadata":"","errorCode":0},{"partition":337,"offset":"4062896","metadata":"","errorCode":0},{"partition":338,"offset":"4365087","metadata":"","errorCode":0},{"partition":339,"offset":"4094258","metadata":"","errorCode":0},{"partition":340,"offset":"4045113","metadata":"","errorCode":0},{"partition":341,"offset":"4208275","metadata":"","errorCode":0},{"partition":342,"offset":"4005344","metadata":"","errorCode":0},{"partition":343,"offset":"4241314","metadata":"","errorCode":0},{"partition":344,"offset":"4241771","metadata":"","errorCode":0},{"partition":345,"offset":"4081372","metadata":"","errorCode":0},{"partition":346,"offset":"4011740","metadata":"","errorCode":0},{"partition":347,"offset":"3970434","metadata":"","errorCode":0},{"partition":348,"offset":"4071969","metadata":"","errorCode":0},{"partition":349,"offset":"3928435","metadata":"","errorCode":0},{"partition":350,"offset":"4347569","metadata":"","errorCode":0},{"partition":351,"offset":"4051655","metadata":"","errorCode":0},{"partition":352,"offset":"4245402","metadata":"","errorCode":0},{"partition":353,"offset":"4044264","metadata":"","errorCode":0},{"partition":354,"offset":"4232091","metadata":"","errorCode":0},{"partition":355,"offset":"4341300","metadata":"","errorCode":0},{"partition":356,"offset":"4277074","metadata":"","errorCode":0},{"partition":357,"offset":"3958455","metadata":"","errorCode":0},{"partition":358,"offset":"4280375","metadata":"","errorCode":0},{"partition":359,"offset":"4256611","metadata":"","errorCode":0},{"partition":360,"offset":"3984831","metadata":"","errorCode":0},{"partition":361,"offset":"4203157","metadata":"","errorCode":0},{"partition":362,"offset":"4178418","metadata":"","errorCode":0},{"partition":363,"offset":"4295067","metadata":"","errorCode":0},{"partition":364,"offset":"4073118","metadata":"","errorCode":0},{"partition":365,"offset":"4189380","metadata":"","errorCode":0},{"partition":366,"offset":"4227132","metadata":"","errorCode":0},{"partition":367,"offset":"4479021","metadata":"","errorCode":0},{"partition":368,"offset":"4135142","metadata":"","errorCode":0},{"partition":369,"offset":"4405261","metadata":"","errorCode":0},{"partition":370,"offset":"4454544","metadata":"","errorCode":0},{"partition":371,"offset":"4146130","metadata":"","errorCode":0},{"partition":372,"offset":"4311696","metadata":"","errorCode":0},{"partition":373,"offset":"4523487","metadata":"","errorCode":0},{"partition":374,"offset":"4984792","metadata":"","errorCode":0},{"partition":375,"offset":"4279559","metadata":"","errorCode":0},{"partition":376,"offset":"4276791","metadata":"","errorCode":0},{"partition":377,"offset":"4218485","metadata":"","errorCode":0},{"partition":378,"offset":"4146955","metadata":"","errorCode":0},{"partition":379,"offset":"4814863","metadata":"","errorCode":0},{"partition":380,"offset":"4667292","metadata":"","errorCode":0},{"partition":381,"offset":"4225096","metadata":"","errorCode":0},{"partition":382,"offset":"4459614","metadata":"","errorCode":0},{"partition":383,"offset":"4354949","metadata":"","errorCode":0},{"partition":384,"offset":"4472844","metadata":"","errorCode":0},{"partition":385,"offset":"4372398","metadata":"","errorCode":0},{"partition":386,"offset":"4570275","metadata":"","errorCode":0},{"partition":387,"offset":"4375883","metadata":"","errorCode":0},{"partition":388,"offset":"4358042","metadata":"","errorCode":0},{"partition":389,"offset":"4515120","metadata":"","errorCode":0},{"partition":390,"offset":"4348105","metadata":"","errorCode":0},{"partition":391,"offset":"4503478","metadata":"","errorCode":0},{"partition":392,"offset":"4481445","metadata":"","errorCode":0},{"partition":393,"offset":"4457423","metadata":"","errorCode":0},{"partition":394,"offset":"4298071","metadata":"","errorCode":0},{"partition":395,"offset":"4379361","metadata":"","errorCode":0},{"partition":396,"offset":"4374381","metadata":"","errorCode":0},{"partition":397,"offset":"4276741","metadata":"","errorCode":0},{"partition":398,"offset":"4410934","metadata":"","errorCode":0},{"partition":399,"offset":"4630902","metadata":"","errorCode":0}]}],"errorCode":0}}
```

**Comment by paambaati** [comment]
@tulios More info!

> the code in master

I don't see this problem, but all reads throw an error from my [`kafkajs-lz4`](https://github.com/indix/kafkajs-lz4) codec, saying "Invalid magic byte 0". This might be a whole different issue.

> and a different group id?

A different group ID seems to fix the issue! The original group ID must've been stale because I was doing a lot of local test runs. Is this expected? I do have a SIGINT handler that calls `disconnect()` on both the consumer and the producer.

**Comment by tulios** [comment]
Nice, at least you can move on. You can reset the offsets of the consumer group and see if it works:
```javascript
await admin.resetOffsets({ groupId, topic }) // latest by default
// await admin.resetOffsets({ groupId, topic, earliest: true })
```

On the LZ4, it's probably not handling the new RecordBatch of Kafka v0.11. I can take a look.

**Comment by paambaati** [comment]
@tulios Thanks! Do you have any guesses about why those `MaxListenersExceededWarning` (see https://github.com/tulios/kafkajs/issues/149#issuecomment-427741795) could occur?

My guess is there's some event emitter that is being set up inside a loop.

**Comment by tulios** [comment]
@paambaati there are a couple of fixes in master for this issue, you can use master disabling the v0.11 APIs:

```javascript
new Kafka({ ... allowExperimentalV011: false })
```

The `MaxListenersExceededWarning` it's a silly bug, it isn't leaking any memory, it is just that node.js has this value hardcoded to 11 and you probably have a higher concurrency. I will create a new issue and fix this right away. It shouldn't cause any issues besides the annoying warning.

If you are producing in parallel you can also consider using the `producer.sendBatch`:

```javascript
await producer.sendBatch({
  topicMessages: <TopicMessages[]>,
  acks: <Number>,
  timeout: <Number>,
  compression: <CompressionTypes>,
})
```

Feel free to re-open this issue if you need.
Thanks again for the report, this is great for new users.

**Comment by tulios** [comment]
For reference, here is the issue about `MaxListenersExceededWarning` #153
